File: 881-boats-to-save-people/881-boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        l, r, counter = 0, len(people) - 1, 0

        while l <= r:
            if people[r] + people[l] <= limit:
                l += 1
            r -= 1
            counter += 1
        return counter

File: 881-boats-to-save-people/881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(), people.end());
        int n = people.size();
        int beg = 0, end = n-1, ans = 0;
        
        while (beg <= end) {
            if (people[beg] + people[end] <= limit)
                beg++;
            end--, ans++;
        }
        return ans;
    }
};